lib/base/utility.cpp

```diff
@@ -1905,3 +1906,48 @@ bool Utility::ComparePasswords(const String& enteredPassword, const String& actu
 
 	return result;
 }
+
+/**
+ * Removes all PerfdataValues from perfdata whose labels don't match any pattern of the filter Array.
+ */
+void Utility::FilterPerfdata(const Array::Ptr& perfdata, const Value& filter)
+{
+	if (!filter.IsObject()) {
+		return;
+	}
+
+	auto filterArray (dynamic_pointer_cast<Array>((Object::Ptr)filter));
+
+	if (!filterArray) {
+		return;
+	}
+
+	auto filterStrings (filterArray->ToSet<String>());
+
+	for (decltype(perfdata->GetLength()) i = 0; i < perfdata->GetLength(); ++i) {
+		auto item (perfdata->Get(i));
+
+		if (!item.IsObject()) {
+			continue;
+		}
+
+		auto itemPerfdataValue (dynamic_pointer_cast<PerfdataValue>((Object::Ptr)item));
+
+		if (!itemPerfdataValue) {
+			continue;
+		}
+
+		auto label (itemPerfdataValue->GetLabel());
+
+		for (auto& pattern : filterStrings) {
+			if (Utility::Match(pattern, label)) {
+				// continue 2;
+				goto NextItem;
+			}
+		}
+
+		perfdata->Remove(i--);
+
+NextItem:;
+	}
```

Comment:
This should be a good place to use `std::remove_if`. That would avoid the need for `goto` here and also has linear complexity (though questionably if anyone would notice a difference here).